cads_adaptors/adaptors/arco.py

```diff
@@ -36,18 +37,43 @@ def _normalise_location(self, request: Request) -> None:
         except (ValueError, TypeError):
             raise InvalidRequest(f"Invalid {location=}. {msg}")
 
-    def _normalise_date(self, request: Request) -> None:
-        date = ensure_list(request.get("date"))
-        if not date:
-            request["date"] = date
-            return
-
-        if len(date) != 1:
+    def _normalise_date(self, request: Request, date_key="date") -> None:
+        date = ensure_list(request.get(date_key))
+        if len(date) == 1:
+            split = sorted(str(date[0]).split("/"))
+            date_range = [split[0], split[-1]]
+        elif len(date) == 2:
+            date_range = date
+        else:
             raise InvalidRequest(
-                "Please specify a single date range using the format yyyy-mm-dd/yyyy-mm-dd."
+                'Please specify a single date range using the format "yyyy-mm-dd/yyyy-mm-dd" or '
+                '["yyyy-mm-dd", "yyyy-mm-dd"].'
+            )
+
+        # Embargo check
+        if "embargo" in self.config and self.config["embargo"]:
+            from datetime import UTC, datetime, timedelta
+
+            from dateutil.parser import parse as dtparse
+
+            embargo = self.config["embargo"]
+            embargo_error_time_format: str = embargo.pop(
+                "error_time_format", "%Y-%m-%d %H:00"
             )
```

Comment:
Do you actually need to pop? If not, I would avoid it. These are the kinds of operations that silently change the internal state of the class and make it unusable twice.

Reply:
Yes, as all the other elements in the dictionary are kwargs for Timedelta